### PR TITLE
Freesurfer ROI Inputs

### PR DESCRIPTION
**Pull request by swashcap**
Asana task: https://app.asana.com/0/37131652076106/167594259597250

Related:

* MRN-Code/coinstac#21
* MRN-Code/freesurfer-parser#1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,6 +21,11 @@ module.exports = {
         return rslt;
       },
       verbose: true,
+      inputs: [{
```
Comment (swashcap) [inline src/index.js]: Eventually, these “question” objects should support different types (“text,” “number,” “date,” etc.). This is a basic `select` for now.

File: src/local.js [hunk 2]
```diff
@@ -14,15 +14,53 @@ module.exports = {
...
+
     const userData = opts.userData;
     // do some file parsing. here, we're actually using pre-processed data...
     // so just add some bias and bail!
     const result = {};
-    const X = userData.files
-    .map((file) => fs.readFileSync(file.filename).toString())
-    .map((strSurf) => new FreeSurfer({ string: strSurf }))
-    .map((surf) => surf.TotalGrayVol); // <== feature to assess!
+    const X = userData.files.map(file => {
```
Comment (swashcap) [inline src/local.js]: Single `Array#map` call to cut down on iteration.

File: src/local.js [hunk 3]
```diff
@@ -14,15 +14,53 @@ module.exports = {
...
+      throw new Error(`Unknown FreeSurfer region in inputs: ${features.toString()}`);
+    }
+
+    /**
+     * @todo: This uses only one feature. Change to use multiple!
+     *
+     * @param {FreeSurfer} freeSurfer
+     * @returns {Number}
+     */
+    function pickFeatures(freeSurfer) {
```
Comment (swashcap) [inline src/local.js]: We’ll need to update this to pick all features. Returning an array makes `run` choke, unfortunately.

Comment (rssk) [inline src/local.js]: bummer

File: src/local.js [hunk 4]
```diff
@@ -14,15 +14,53 @@ module.exports = {
...
+      return freeSurfer[features[0]];
```
Comment (rssk) [inline src/local.js]: So this is just the first 'element' that is returned by the select plugin?

File: src/local.js (hunk 2 above)
Comment (rssk) [inline src/local.js]: 👍 👍

File: src/local.js (hunk 4 above)
Comment (swashcap) [inline src/local.js]: Yeah. It _should_ be stored in something like:

```js
{
  "pluginState": {
    "inputs": [
      ["Left-Hippocampus", "Right-Hippocampus", "Thalamus"]
    ],
    "step": {
      // ...
    }
  }
}
```

We’ll only pick the first element, `Left-Hippocampus`.

**Comment by swashcap** [comment]
Merging this and bumping to `3.0.0` as per discussion with @rssk.

**Pull request merged**